core/cog-utils.h

```diff
@@ -97,7 +97,12 @@ size_t cog_strndup(const char src[], size_t len, char **p_dest);
  * @param ... variadic arguments to be matched to `fmt` specifiers
  * @return length of copied string on success, -1 on failure
  */
-size_t cog_asprintf(char **strp, const char fmt[], ...);
+size_t cog_asprintf(char **strp, const char fmt[], ...)
+#if defined(__MINGW32__) || (defined (__GNUC__) && __GNUC__ > 4 ? true : __GNUC_PATCHLEVEL__ >= 4) || defined(__USE_MINGW_ANSI_STDIO)
+    __attribute__((format(gnu_printf, 2, 3)));
+#else
+;
+#endif
```

Comment:
Since this is being used in many situations you may consider adding a macro for it instead:
```c
#if defined(__MINGW32__) || (defined (__GNUC__) && __GNUC__ > 4 ? true : __GNUC_PATCHLEVEL__ >= 4) || defined(__USE_MINGW_ANSI_STDIO)
#  define PRINTF_LIKE(a, b) __attribute__((format(gnu_printf, a, b)))
#else
#  define PRINTF_LIKE(a, b)
#endif
```
And then use it like so:
```c
size_t cog_asprintf(char **strp, const char fmt[], ...) PRINTF_LIKE(2, 3);
```